File: TTdsa-java/Exercise/src/com/aacirq/string/N02.java
package com.aacirq.string;

public class N02 {
    public static void main(String[] args) {
        N02 sol = new N02();
        // "We%20Are%20Happy"
        System.out.println(sol.replaceSpace("We Are Happy"));
    }

    public String replaceSpace(String s) {
        StringBuilder str = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                str.append("%20");
            } else {
                str.append(s.charAt(i));
            }
        }
        return str.toString();
    }
}

File: TTdsa-cpp/Exercise/string/N02.cpp
//
// Created by renxin on 2021/6/9.
//

#include "ttdsa.h"

class Solution {
public:
    string replaceSpace(string s) {
        int count = 0;
        int len = s.size();
        for (auto c : s) {
            if (c == ' ') count++;
        }
        s.resize(len + 2 * count);
        int i = s.size() - 1;
        int j = len - 1;
        while (j < i) {
            if (s[j] != ' ') {
                s[i] = s[j];
                i--;
            } else {
                s[i] = '0';
                s[i - 1] = '2';
                s[i - 2] = '%';
                i -= 3;
            }
            j--;
        }
        return s;
    }
};

int main() {
    Solution sol;

    // "We%20Are%20Happy"
    cout << sol.replaceSpace("We Are Happy");
    return 0;
}